### DISCUSSION TITLE
Howto refresh/clean out the database periodically

### DISCUSSION THREAD

**Discussion by KingJulianS** [GothenburgBitFactory/timewarrior#517] [Q&A] [answered]
Hi, how can I refresh/cleanup/remove older data from the timewarrior data to start fresh? Now that I understand it better and have established a new tag naming convention I'd like to use, I would like to remove some chunks Of older data. 

Thanks in advance

**Comment by sruffell** [comment, marked as answer]
Hello. Currently, the easiest way to delete old data is to remove data files from the database.

The output of `timew diag` will show you where timewarrior currently has the database:

```sh
$ timew diag | grep Database:
       Database: xxx
 ```
 
In a directory named `data` under the database directory is where timewarrior currently stores interval data.  There are the YYYY-mm.data files that store intervals, tags.data which tracks known tags to report if a tag is new ,and undo.data to support the `timew undo` operation.

If tags.data is missing, timew will recreate it from any existing YYYY-mm.data files, and the undo is only needed to support `timew undo`.  So you can delete any older YYYY-mm.data files you want, then delete the tags.data and undo.data files and you should be good to go.

To completely start fresh you can delete everything under data.

**Reply by KingJulianS** [reply]
Thank you so much for that detailed explanation. Much appreciated

**Reply by KingJulianS** [reply]
Hi, thanks again. I created a directory bkup_ccyy and then $ mv *.data bkup_2022 ,  and all seems good to go.
